packages/terminal/src/browser/terminal-frontend-module.ts

```diff
@@ -66,6 +70,22 @@ export default new ContainerModule(bind => {
         }
     }));
 
+    bind(TerminalSearchWidgetFactory).toDynamicValue(ctx => (terminal: Terminal, node: Element, terminalWdgId: string) => {
+        if (ctx.container.isBoundNamed(TerminalSearchWidget, terminalWdgId)) {
+            return ctx.container.getNamed(TerminalSearchWidget, terminalWdgId);
+        }
+
+        const container = new Container({ defaultScope: 'Singleton' });
+        container.bind(Terminal).toConstantValue(terminal);
+        container.bind(Element).toConstantValue(node);
+        container.bind(TerminalSearchWidget).toSelf().inSingletonScope();
+
+        const widget = container.get(TerminalSearchWidget);
+        ctx.container.bind(TerminalSearchWidget).toConstantValue(widget).whenTargetNamed(terminalWdgId);
```

Comment:
Why don't you create a new instance each time?

Reply:
ok, i see it is scoped with `terminalWdgId`

please use whole worlds btw `terminalWidgetId`, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#whole-words-names

Reply:
Could not it be a part of a terminal widget factory instead? There is already a new DI container for each terminal. We could bind `TerminalSearchWidgetFactory` in it, so it will create always only one instance for this terminal. 

Also the api will be good on the `TerminalWidget` to access the search box, like : `terminalWidget.search`, so in the keybinding context instead of inserting `TerminalSearchWidgetFactory`, you get an active terminal, access its search and check whether it is hidden or not.